packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -257,6 +258,20 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-Policy-V1-03
+    PropertiesTypeObjectNoDefinition: {
+      description:
+        "Properties with type:object that dont reference a model definition are not allowed. ARM doesnt allow generic type definitions as this leads to bad customer experience.",
+      severity: "error",
+      message: "{{error}}",
+      resolved: true,
+      formats: [oas2],
+      given: "$.definitions..[?((@property === 'type' && @ ==='object' || @ ===''))]^",
```

Comment:
I've noticed with JSON Path when you use the `^` to get the parent after using a filter `[? ... ]`, the filter has no effect. So, you should be able to remove the filter and parent and have the same results.

Might need to double check my syntax here:

```suggestion
      given: "$.definitions..",
```

Reply:
Brett, I tried removing this and checked the test it's working fine but I when I debug the code there are extra checks that's happening which are not required. 
This Given expression is simplifying the unnecessary checks.